layers.go

```diff
@@ -1200,16 +1197,19 @@ func (r *layerStore) PutAdditionalLayer(id string, parentLayer *Layer, names []s
 		r.byuncompressedsum[layer.UncompressedDigest] = append(r.byuncompressedsum[layer.UncompressedDigest], layer.ID)
 	}
 	if err := r.saveFor(layer); err != nil {
-		if err2 := r.driver.Remove(id); err2 != nil {
-			logrus.Errorf("While recovering from a failure to save layers, error deleting layer %#v: %v", id, err2)
+		if e := r.Delete(layer.ID); e != nil && !errors.Is(e, os.ErrNotExist) {
```

Comment:
The error from `Delete` is ErrLayerUnknown .

Reply:
Right, removing that `errors.Is()` call.